include/url.h

```diff
@@ -0,0 +1,86 @@
+#ifndef URL_CPP_H
+#define URL_CPP_H
+
+#include <stdexcept>
+#include <string>
+#include <vector>
+#include <unordered_map>
+#include <unordered_set>
+
+namespace Url {
+
+    struct UrlParseException : public std::logic_error {
+        UrlParseException(const std::string& message): std::logic_error(message) {}
+    };
+
+    struct Url {
+        /* Character classes */
+        const static std::string GEN_DELIMS;
+        const static std::string SUB_DELIMS;
+        const static std::string ALPHA;
+        const static std::string DIGIT;
+        const static std::string UNRESERVED;
+        const static std::string RESERVED;
+        const static std::string PCHAR;
+        const static std::string PATH;
+        const static std::string QUERY;
+        const static std::string FRAGMENT;
+        const static std::string USERINFO;
+        const static std::string HEX;
+        const static std::string SCHEME;
+
+        Url(const std::string& url);
+
+        /**************************
+         * Component-wise access. *
+         **************************/
+        const std::string& getScheme() const {
```

Comment:
This class is a data value type like `std::complex`, so I think having accessors is fine, but maybe you mean the literal use of `get` in the method name. It's pretty common not to include the word "get"in an accessor in C++. That's more of a Java-ism.